src/AspNet.Security.OAuth.Vkontakte/VkontakteAuthenticationHandler.cs

```diff
@@ -45,23 +50,24 @@ public VkontakteAuthenticationHandler(HttpClient client)
 
                 throw new HttpRequestException("An error occurred while retrieving the user profile.");
             }
+            
+            var payloadContainer = JObject.Parse(await response.Content.ReadAsStringAsync());
+            var payload = payloadContainer["response"].First as JObject;
 
-            var payload = JObject.Parse(await response.Content.ReadAsStringAsync());
-            var user = (JObject) payload["response"][0];
+            if (tokens.Response["email"] != null)
```

Comment:
Looks like a behavior change? If so, please revert it and add a new ticket so we can change that in both the 1.0 and 2.0 versions.